src/main/java/ru/yandex/clickhouse/jdbcbridge/db/clickhouse/ClickHouseConverter.java

```diff
@@ -62,16 +51,28 @@
         MAP = Collections.unmodifiableMap(map);
     }
 
-    public static ClickHouseDataType getBySQLType(int type) throws SQLException {
-        return getInstructionBySQLType(type).clickHouseDataType;
+    public static ClickHouseDataType getBySQLType(int type, int precision, int scale) throws SQLException {
+        return getInstructionBySQLType(type, precision, scale).clickHouseDataType;
     }
 
-    public static ExtractorConverter<?> getSerializerBySQLType(int type) throws SQLException {
-        return getInstructionBySQLType(type).serializer;
+    public static ExtractorConverter<?> getSerializerBySQLType(int type, int precision, int scale) throws SQLException {
+        return getInstructionBySQLType(type, precision, scale).serializer;
     }
 
 
-    private static MappingInstruction<?> getInstructionBySQLType(int sqlType) throws SQLException {
+    private static MappingInstruction<?> getInstructionBySQLType(int sqlType, int precision, int scale) throws SQLException {
+
+        // for decimal types
+        if (sqlType == NUMERIC){
+            if (precision <=9 ){
+                return new MappingInstruction<>(Decimal, ResultSet::getBigDecimal, (i, s) -> s.writeDecimal32(i, scale));
```

Comment:
I wonder do you allocate Instruction each time, instead of returning a reference from the Map? May be generate a Map for all possible dimensions of decimal?

Reply:
all options of decimal was added in MAP

key in map was changed. for key I created new class - TypeWithScaleAndPrecision